src/main/java/uk/gov/justice/hmpps/datacompliance/services/deletion/DeletionService.java

```diff
@@ -0,0 +1,86 @@
+import uk.gov.justice.hmpps.datacompliance.events.publishers.dto.OffenderDeletionCompleteEvent.Booking;
+import uk.gov.justice.hmpps.datacompliance.events.publishers.dto.OffenderDeletionCompleteEvent.OffenderWithBookings;
+import uk.gov.justice.hmpps.datacompliance.events.publishers.sns.OffenderDeletionCompleteEventPusher;
+import uk.gov.justice.hmpps.datacompliance.events.publishers.sqs.DataComplianceEventPusher;
+import uk.gov.justice.hmpps.datacompliance.repository.jpa.model.referral.OffenderDeletionReferral;
+import uk.gov.justice.hmpps.datacompliance.repository.jpa.model.referral.ReferredOffenderBooking;
+import uk.gov.justice.hmpps.datacompliance.repository.jpa.repository.referral.OffenderDeletionReferralRepository;
+import uk.gov.justice.hmpps.datacompliance.utils.TimeSource;
+
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.stream.Collectors.groupingBy;
+import static uk.gov.justice.hmpps.datacompliance.repository.jpa.model.referral.ReferralResolution.ResolutionStatus.DELETED;
+import static uk.gov.justice.hmpps.datacompliance.repository.jpa.model.referral.ReferralResolution.ResolutionStatus.DELETION_GRANTED;
+import static uk.gov.justice.hmpps.datacompliance.utils.Exceptions.illegalState;
+
+@Slf4j
+@Service
+@Transactional
+@AllArgsConstructor
+public class DeletionService {
+
+    private final TimeSource timeSource;
+    private final OffenderDeletionReferralRepository referralRepository;
+    private final OffenderDeletionCompleteEventPusher deletionCompleteEventPusher;
+    private final DataComplianceEventPusher deletionGrantedEventPusher;
+
+    public void grantDeletion(final OffenderDeletionReferral referral) {
+        deletionGrantedEventPusher.grantDeletion(new OffenderNumber(referral.getOffenderNo()), referral.getReferralId());
+    }
+
+    public void handleDeletionComplete(final OffenderDeletionCompleteEvent event) {
+
+        final var referral = referralRepository.findById(event.getReferralId())
+                .orElseThrow(illegalState("Cannot retrieve referral record for id: '%s'", event.getReferralId()));
+
+        checkState(Objects.equals(event.getOffenderIdDisplay(), referral.getOffenderNo()),
+                "Offender number '%s' of referral '%s' does not match '%s'",
+                referral.getOffenderNo(), referral.getReferralId(), event.getOffenderIdDisplay());
+
+        recordDeletionCompletion(referral);
+        publishDeletionCompleteEvent(referral);
+    }
+
+    private void recordDeletionCompletion(final OffenderDeletionReferral referral) {
+
+        log.info("Updating destruction log with deletion confirmation for: '{}'", referral.getOffenderNo());
+
+        final var referralResolution = referral.getReferralResolution()
+                .filter(resolution -> resolution.isType(DELETION_GRANTED))
+                .orElseThrow(illegalState("Referral '%s' does not have expected resolution type", referral.getReferralId()));
```

Comment:
It can probably derived from the service name but perhaps include the actual resolution type value or values? meh.